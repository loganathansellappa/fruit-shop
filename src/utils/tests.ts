import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

export const mockResource = () => {
  return {
    discounts: [
      {
        id: 1,
        valid: true,
        type: 'Quantity',
        product_type: 'Apple',
        value: 2,
        reward: 1,
      },
    ],
    prodWithDiscount: {
      id: 2,
      type: 'Apple',
      imageUrl:
        'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
      title: 'Apple - 2',
      description:
        'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
      price: 1,
    },
    prodWithoutDiscount: {
      id: 2,
      type: 'Banana',
      imageUrl:
        'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
      title: 'Banana',
      description:
        'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
      price: 2,
    },
    listResponse: {
      products: [
        {
          id: 1,
          type: 'Banana',
          imageUrl:
            'https://avectime.com/grocery&gourmet/agriculture/brands/images/gros_michel_banana/gros_michel_banana1.jpg',
          title: 'Banana - 1',
          description:
            'BananaLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
        },
        {
          id: 2,
          type: 'Apple',
          imageUrl:
            'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
          title: 'Apple - 2',
          description:
            'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 1,
        },
        {
          id: 3,
          type: 'Pineapple',
          imageUrl:
            'https://helios-i.mashable.com/imagery/articles/05W5DssM7oLPbBjiU4ZY6ob/hero-image.fill.size_1248x702.v1645798494.jpg',
          title: 'Pineapple - 3',
          description:
            'PineappleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 5,
        },
        {
          id: 4,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 4',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 25,
        },
        {
          id: 5,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 5',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
        },
        {
          id: 6,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 6',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 29,
        },
        {
          id: 7,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 7',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 15,
        },
        {
          id: 8,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 8',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 9,
        },
        {
          id: 9,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 9',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 17,
        },
        {
          id: 10,
          type: 'Other',
          imageUrl:
            'https://img.pixers.pics/pho_wat(s3:700/FO/78/15/78/42/700_FO78157842_1e837a4bfb4e3bdabed3afa8ae0e4361.jpg,700,465,cms:2018/10/5bd1b6b8d04b8_220x50-watermark.png,over,480,415,jpg)/seitenschlaferkissen-frische-fruchte-mixed-fruits-background-dieting-gesunde-ernahrung.jpg.jpg',
          title: 'Other - 10',
          description:
            'OtherLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 21,
        },
      ],
      discounts: [
        {
          id: 53,
          valid: true,
          type: 'Quantity',
          product_type: 'Apple',
          value: 2,
          reward: 1,
        },
        {
          id: 54,
          valid: true,
          type: 'Price',
          product_type: 'Pineapple',
          value: 2,
          reward: 3,
        },
      ],
      total: 52,
    },
    cart: {
      products: {
        Banana: {
          id: 1,
          type: 'Banana',
          imageUrl:
            'https://avectime.com/grocery&gourmet/agriculture/brands/images/gros_michel_banana/gros_michel_banana1.jpg',
          title: 'Banana - 1',
          description:
            'BananaLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 2,
          totalQuantity: 4,
          totalPrice: 8,
          discountPrice: 0,
          discountQuantity: 0,
          finalQuantity: 4,
          finalPrice: 8,
        },
        Apple: {
          id: 2,
          type: 'Apple',
          imageUrl:
            'https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg',
          title: 'Apple - 2',
          description:
            'AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat lorem, et vehicula enim. Nam erat nunc, efficitur eget nibh at, euismod molestie justo. Nunc dictum euismod lacus nec varius. Sed efficitur dui et magna elementum pharetra. Proin ipsum mauris, pretium id tellus at, ultrices maximus metus. Interdum et malesuada fames ac ante ipsum primis in faucibus. In arcu dui, volutpat in sem vehicula, hendrerit lacinia elit. Nullam tristique augue non tempor efficitur. Nulla iaculis nibh at tortor pharetra, sit amet ornare lectus auctor. Sed nec sapien at est fermentum posuere. Fusce suscipit tristique velit, et consectetur nunc dignissim nec. Sed id ligula ut ante gravida ornare. Suspendisse eu odio lacus. Proin rhoncus convallis vehicula.',
          price: 1,
          totalQuantity: 4,
          totalPrice: 4,
          discountPrice: 0,
          discountQuantity: 2,
          finalQuantity: 6,
          finalPrice: 4,
        },
      },
      totalQuantity: 10,
      totalPrice: 12,
      totalUserQuantity: 8,
    },
  };
};
