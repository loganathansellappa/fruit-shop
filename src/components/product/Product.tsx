import { Button, Grid, Typography } from '@material-ui/core';
import { Wrapper } from './product.styles';
import { Discount, ProductDetail } from '../../@types/Product';
import { truncateString } from '../../utils/HelperUtils';
import { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';

type Props = {
  product: ProductDetail;
  owner: boolean;
  discount?: Discount;
  handleAddToCart: (a: ProductDetail, b: Discount) => void;
};

const Product = ({ product, handleAddToCart, owner, discount }: Props) => {
  const action = useCallback(
    (owner: boolean) => {
      const buttonStyle = {
        backgroundColor: 'green',
        color: '#fff',
        borderRadius: '0 0 20px 20px',
      };
      if (!owner) {
        return (
          <Button
            onClick={(e) => {
              handleAddToCart(product, discount as Discount);
            }}
            style={{ ...buttonStyle }}
          >
            Add to Cart
          </Button>
        );
      } else {
        const navigate = useNavigate();
        return (
          <Button
            onClick={() => navigate(`/admin/products/${product.id}`)}
            style={{ ...buttonStyle }}
          >
            View More
          </Button>
        );
      }
    },
    [owner]
  );

  return (
    <Wrapper>
      <img src={product.imageUrl} alt={product.title} />
      <Grid>
        <Typography noWrap variant={'h4'}>
          {product.title}
        </Typography>
        <Typography variant={'body1'}>
          {truncateString(product.description)}
        </Typography>
        <Typography noWrap variant={'h4'}>
          â‚¬{product.price}
        </Typography>
      </Grid>
      {action(owner)}
    </Wrapper>
  );
};

export default Product;
