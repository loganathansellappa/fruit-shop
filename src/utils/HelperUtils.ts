import { Discount, ProductDetail } from '~/@types/Product';

export const apiService = () => `${process.env.SHOP_URL}`;
export const truncateString = (string = '', maxLength = 20) =>
  string.length > maxLength ? string.split(' ').slice(0, 20).join(' ') : string;
export const getDiscount = (product: ProductDetail, discounts: Discount[]) =>
  discounts.find((d) => product.type === d.product_type);
