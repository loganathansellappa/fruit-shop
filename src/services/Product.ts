import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { GetProductsResponse, ProductDetail } from '../@types/Product';
import { apiService } from '../utils/HelperUtils';
import { Buffer } from 'buffer';

export const productApi = createApi({
  reducerPath: 'productApi',
  baseQuery: fetchBaseQuery({ baseUrl: apiService() }),
  tagTypes: ['Products'],

  endpoints: (builder) => ({
    getProducts: builder.query<GetProductsResponse | ProductDetail, string>({
      query: (url) => ({
        url,
        headers: {
          contentType: 'application/json',
        },
      }),
      providesTags: ['Products'],
    }),
    editProduct: builder.mutation({
      query: (product) => ({
        url: `/products/${product.id}?`,
        method: 'PUT',
        headers: {
          Authorization: `Basic ${Buffer.from('admin:admin').toString(
            'base64'
          )}`,
        },
        body: product,
      }),
      invalidatesTags: ['Products'],
    }),
  }),
});

export const { useGetProductsQuery, useEditProductMutation } = productApi;
