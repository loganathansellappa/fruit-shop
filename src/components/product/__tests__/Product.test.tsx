import { mockResource, renderWithRouter } from '../../..//utils/tests';
import Product from '../Product';

describe('Product', () => {
  let element: { container: HTMLElement };
  const handleAddToCart = jest.fn();
  beforeEach(() => {
    element = renderWithRouter(
      <Product
        product={mockResource().prodWithoutDiscount}
        discount={mockResource().discounts[0]}
        handleAddToCart={handleAddToCart}
      />
    );
  });

  test('renders component for user', () => {
    expect(element.container.firstChild).toMatchSnapshot();
    expect(element.container.textContent).toContain(
      mockResource().prodWithoutDiscount.title
    );
    expect(element.container.textContent).toContain('Add to Cart');
  });

  test('renders component for owner', () => {
    element = renderWithRouter(
      <Product
        product={mockResource().prodWithoutDiscount}
        discount={mockResource().discounts[0]}
        handleAddToCart={handleAddToCart}
        owner={true}
      />
    );
    expect(element.container.firstChild).toMatchSnapshot();
    expect(element.container.textContent).toContain(
      mockResource().prodWithoutDiscount.title
    );
    expect(element.container.textContent).toContain('View More');
  });
});
