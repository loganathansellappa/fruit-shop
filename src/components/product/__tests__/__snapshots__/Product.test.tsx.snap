// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders component for owner 1`] = `
<div
  class="sc-bczRLJ fPmJiV"
>
  <img
    alt="Banana"
    src="https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg"
  />
  <div
    class="MuiGrid-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap"
    >
      Banana
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat
    </p>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap"
    >
      €
      2
    </h4>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    style="background-color: green; color: rgb(255, 255, 255); border-radius: 0 0 20px 20px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      View More
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`Product renders component for user 1`] = `
<div
  class="sc-bczRLJ fPmJiV"
>
  <img
    alt="Banana"
    src="https://images.pexels.com/photos/672101/pexels-photo-672101.jpeg"
  />
  <div
    class="MuiGrid-root"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap"
    >
      Banana
    </h4>
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      AppleLorem ipsum dolor sit amet, consectetur adipiscing elit. Sed bibendum ante purus, id fermentum nulla commodo vehicula. Nulla id volutpat
    </p>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-noWrap"
    >
      €
      2
    </h4>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text"
    style="background-color: green; color: rgb(255, 255, 255); border-radius: 0 0 20px 20px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Add to Cart
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
